src/tools/compiletest/src/common.rs

```diff
@@ -215,15 +215,22 @@ pub struct Config {
     /// `None` then these tests will be ignored.
     pub run_clang_based_tests_with: Option<String>,
 
-    /// The directory containing the tests to run
-    pub src_base: PathBuf,
-
-    /// The directory where programs should be built
-    pub build_base: PathBuf,
-
-    /// The directory containing the compiler sysroot
-    pub sysroot_base: PathBuf,
-
+    /// Path to directory containing rust-lang/rust sources.
+    pub src_root: PathBuf,
+    /// Path to the directory containg the test suite sources. Expected to be a subdirectory of
+    /// `src_root`.
+    pub src_test_suite_root: PathBuf,
+
+    /// Bootstrap build directory.
+    pub build_root: PathBuf,
+    /// Bootstrap test suite build directory. Expected to be a subdirectory of `build_root`.
+    pub build_test_suite_root: PathBuf,
```

Comment:
These are terribly named, and looking at the runtest impls, it has confused multiple people. `src_base` is not sources root, it's the directory `$SRC_ROOT/tests/<test-suite>/`. `build_base` is not build directory root, it's `$BUILD/<host-triple>/test/<test-suite>`.